crates/chain/src/keychain/txout_index.rs

```diff
@@ -367,60 +525,70 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     ///
     /// This will panic if the given `keychain`'s descriptor does not exist.
     pub fn unbounded_spk_iter(&self, keychain: &K) -> SpkIterator<Descriptor<DescriptorPublicKey>> {
-        SpkIterator::new(
-            self.keychains
-                .get(keychain)
-                .expect("keychain does not exist")
-                .clone(),
-        )
+        let descriptor = self
+            .descriptor_of_keychain(keychain)
+            .expect("Keychain must exist")
+            .1
+            .clone();
+        SpkIterator::new(descriptor)
     }
 
     /// Get unbounded spk iterators for all keychains.
     pub fn all_unbounded_spk_iters(
         &self,
     ) -> BTreeMap<K, SpkIterator<Descriptor<DescriptorPublicKey>>> {
-        self.keychains
+        self.keychains_to_descriptors
             .iter()
-            .map(|(k, descriptor)| (k.clone(), SpkIterator::new(descriptor.clone())))
+            .map(|(k, (_, descriptor))| (k.clone(), SpkIterator::new(descriptor.clone())))
             .collect()
     }
 
     /// Iterate over revealed spks of all keychains.
     pub fn revealed_spks(&self) -> impl DoubleEndedIterator<Item = (K, u32, &Script)> + Clone {
-        self.keychains.keys().flat_map(|keychain| {
+        self.keychains_to_descriptors.keys().flat_map(|keychain| {
             self.revealed_keychain_spks(keychain)
                 .map(|(i, spk)| (keychain.clone(), i, spk))
         })
     }
 
     /// Iterate over revealed spks of the given `keychain`.
+    ///
+    /// # Panics
```

Comment:
Is there really any way for this (and below functions) to panic? If the logic of `add_keychain()` is correct (which I think it is) there shouldn't be a way. The `.expect("Must exist")` lines should only fail during development and testing if someone breaks the logic.